compiler/src/iree/compiler/Dialect/LinalgExt/Utils/Utils.cpp

```diff
@@ -445,20 +445,40 @@ getIGEMMGenericConvDetails(linalg::LinalgOp linalgOp) {
   auto outputShape = outputType.getShape();
   auto indexingMaps = linalgOp.getIndexingMapsArray();
   auto filterMap = indexingMaps[1];
+  auto outputMap = indexingMaps[2];
 
   SmallVector<int64_t> reductionDims;
   for (auto iter : llvm::enumerate(linalgOp.getIteratorTypesArray())) {
     if (linalg::isReductionIterator(iter.value())) {
       reductionDims.push_back(iter.index());
     }
   }
+
+  bool isOutputChannelFirst = false;
+  auto outputChannelPos = convDims.outputChannel;
+  auto outputImagePos = convDims.outputImage;
+
+  std::optional<int64_t> outputChannelLastDim = outputMap.getResultPosition(
+      getAffineDimExpr(outputChannelPos.back(), outputMap.getContext()));
+  std::optional<int64_t> outputImageFirstDim = outputMap.getResultPosition(
+      getAffineDimExpr(outputImagePos[0], outputMap.getContext()));
+  if (!outputImageFirstDim || !outputChannelLastDim) {
+    LDBG("output image or output channel dim not found in output.");
+    return failure();
+  }
+  if (outputChannelLastDim.value() < outputImageFirstDim.value())
+    isOutputChannelFirst = true;
+
   SmallVector<int64_t> filterkPos;
   for (auto reductionDim : reductionDims) {
     std::optional<int64_t> maybeDim = filterMap.getResultPosition(
         getAffineDimExpr(reductionDim, filterMap.getContext()));
     filterkPos.push_back(maybeDim.value());
   }
-  // group together adjacent reduction dimensions in the filter
+  // group together adjacent reduction dimensions in the filter.
+  // First we want to sort the dims as the look up from the filterMap
+  // can place the dims in arbitarty order.
+  std::sort(filterkPos.begin(), filterkPos.end());
```

Comment:
I don't quite get why a sort is needed here. 

Let's say this is a 2d convolution, and of filter layout `fchw`. The reduction dimensions are `chw` collapsed together. Let's assume the filter map is associated as:
 - d4 -> h
 - d5 -> w
 - d6 -> c
By doing a sort, you re-arranged the collapse of filter map (to  (d4, d5, d6), or (h, w, c)) in a reverse ordering compared to its physical layout (c, h, w). I assume this will disrupt the memory access pattern compared with you just taking the original ordering as-is from affine map result, which should be (d6, d4, d5).

Reply:
Okay I think I've got myself confused on the functionality of this `filterkPos` vector and thought this has to do with how we collapse the filter dimension together. 

Effectively in your example, (d3, d5, d6, d4) will still be the accessing order of the filter tensor, just with a different affine symbol representing each dimension (and we don't alter that sequence in this pass at all). A sort can be handy in the subsequent processing. It looks the subsequent code require the neighboring dimensions to be in sorted order just so that you can tell if the reduction dimension is contiguous or split.